docs/DeveloperGuide.md

```diff
@@ -1,29 +1,194 @@
 # Developer Guide
 
+## Table of Contents
+* [Acknowledgements](#acknowledgements)
+* [Design & Implementation](#design--implementation)
+* * [Categorising the different books by their genres](#categorising-the-different-books-by-their-genres)
+* * [BookList Class Component](#booklist-class-component)
+* * [ParserMain Class Component](#parsermain-class-component)
+* * [FileStorage Class Component](#filestorage-class-component)
+* [Product Scope](#product-scope)
+* * [Target user profile](#target-user-profile)
+* * [Value proposition](#value-proposition)
+* [User Stories](#user-stories)
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+Reference to AB-3 Developer Guide
+* [Source URL](https://se-education.org/addressbook-level3/DeveloperGuide.html#documentation-logging-testing-configuration-dev-ops)
+* Used as template to structure this DeveloperGuide
+* Reference to AB-3 diagrams code
+
+Reference to AB-3 diagrams code
+* [Source URL](https://github.com/se-edu/addressbook-level3/tree/master/docs/diagrams)
+* Used as reference to understand PlantUML syntax
+
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
```

Comment:
Perhaps you could remove the placeholder text so that it's clearer where the Design and Implementation section begins